src/traits.rs

```diff
@@ -656,14 +656,44 @@ pub trait PrimInt
     + BitAnd<Output=Self>
     + BitOr<Output=Self>
     + BitXor<Output=Self>
-    + Shl<usize, Output=Self>
-    + Shr<usize, Output=Self>
+    + Shl<u32, Output=Self>
+    + Shr<u32, Output=Self>
     + CheckedAdd<Output=Self>
     + CheckedSub<Output=Self>
     + CheckedMul<Output=Self>
     + CheckedDiv<Output=Self>
+    + Sub<Output=Self>
     + Saturating
 {
+    type Signed : PrimInt;
+    type Unsigned : PrimInt;
+
+    /// Transmutes the integer into an unsigned integer of the
+    /// same size (loss less).
+    ///
+    /// # Examples
+    ///
+    /// ~~~
+    /// use num::traits::PrimInt;
+    ///
+    /// assert_eq!((8i32).bits_to_unsigned().bits_to_signed(), 8i32);
+    /// assert_eq!((-1i32).bits_to_unsigned().bits_to_signed(), -1i32);
+    /// ~~~
+    fn bits_to_unsigned(self) -> Self::Unsigned;
+
+    /// Transmutes an integer into a signed integer of the
+    /// same size (loss less).
+    ///
+    /// # Examples
+    ///
+    /// ~~~
+    /// use num::traits::PrimInt;
+    ///
+    /// assert_eq!((8i32).bits_to_unsigned().bits_to_signed(), 8i32);
+    /// assert_eq!((-1i32).bits_to_unsigned().bits_to_signed(), -1i32);
+    /// ~~~
+    fn bits_to_signed(self) -> Self::Signed;
+ 
```

Comment:
I'm still not sure we want these, but if so, I'd call them `to_unsigned` and `to_signed`.